packages/atomic/src/components/product-listing/atomic-product-listing-pager/atomic-product-listing-pager.tsx

```diff
@@ -0,0 +1,127 @@
+import {SearchStatus, SearchStatusState} from '@coveo/headless';
+import {
+  buildController,
+  buildPager,
+  Pager,
+  PagerState,
+  ProductListingEngine,
+} from '@coveo/headless/product-listing';
+import {Component, Event, EventEmitter, h, Prop, State} from '@stencil/core';
+import ArrowLeftIcon from '../../../images/arrow-left-rounded.svg';
+import ArrowRightIcon from '../../../images/arrow-right-rounded.svg';
+import {
+  FocusTarget,
+  FocusTargetController,
+} from '../../../utils/accessibility-utils';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {PagerCommon} from '../../common/pager/pager-common';
+import {ProductListingBindings} from '../atomic-product-listing-interface/atomic-product-listing-interface';
+
+/**
+ * The `atomic-product-listing-pager` provides buttons that allow the end user to navigate through the different result pages.
+ *
+ * @part buttons - The list of the next/previous buttons and page-buttons.
+ * @part page-buttons - The list of page buttons.
+ * @part page-button - The page button.
+ * @part active-page-button - The active page button.
+ * @part previous-button - The previous button.
+ * @part next-button - The next button.
+ * @part previous-button-icon - Icon of the previous button.
+ * @part next-button-icon - Icon of the next button.
```

Comment:
Similar to my comment on your other PR, I would mark this one `@internal` for the time being.